src/group/mls_group/mod.rs

```diff
@@ -200,23 +194,33 @@ impl Api for MlsGroup {
 
     // Encrypt/Decrypt MLS message
     fn encrypt(&mut self, mls_plaintext: MLSPlaintext) -> MLSCiphertext {
-        let context = &self.get_context();
-        MLSCiphertext::new_from_plaintext(
-            &mls_plaintext,
-            &self.get_ciphersuite().clone(),
-            &mut self.astree,
-            &self.epoch_secrets,
-            context,
-        )
+        let mut astree = self.astree.borrow_mut();
+        let generation = astree.get_generation(mls_plaintext.sender.sender);
+        let application_secrets = astree
+            .get_secret(&self.ciphersuite, mls_plaintext.sender.sender, generation)
+            .unwrap();
+        MLSCiphertext::new_from_plaintext(&mls_plaintext, &self, generation, &application_secrets)
     }
+
     fn decrypt(&mut self, mls_ciphertext: MLSCiphertext) -> MLSPlaintext {
-        let context = &self.get_context();
+        let tree = self.tree.borrow();
```

Comment:
We might use the roster() function elsewhere going forward, any reason it should be inline here?

Reply:
This is the only place where the `roster` function was used. Also, it's not usable as is (`self` is borrowed immutable and it's returning borrowed values).
I think I'd keep it inlined here for this iteration and see how we can pull it out once we need it elsewhere.

Reply:
Fair enough